hoomd/pytest/test_simulation.py

```diff
@@ -197,6 +197,38 @@ def test_state_from_gsd(simulation_factory, get_snapshot,
         assert box == sim.state.box
         assert_equivalent_snapshots(snap, sim.state.snapshot)
 
+@skip_gsd
+def test_state_from_snapshot(simulation_factory, get_snapshot,
+                        device, state_args, tmp_path):
+    snap_params, nsteps = state_args
+
+    d = tmp_path / "sub"
+    d.mkdir()
+    filename = d / "temporary_test_file.gsd"
+    with gsd.hoomd.open(name=filename, mode='wb+') as file:
```

Comment:
In your tests there is no reason to write to a file since you only use the stored `gsd.hoomd.Snapshot` objects.